From 0ffee54ecde769ac2230ed8e56b260aa48101ef6 Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Wed, 4 Nov 2015 12:27:03 -0500
Subject: [PATCH 6/7] ARM: dts: k2l: add device binding for crypto accelerator

This patch enables K2L HW crypto accelerator by adding device binding
for it.

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
---
 arch/arm/boot/dts/k2l-netcp.dtsi | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm/boot/dts/k2l-netcp.dtsi b/arch/arm/boot/dts/k2l-netcp.dtsi
index debcbc4..60d1311 100644
--- a/arch/arm/boot/dts/k2l-netcp.dtsi
+++ b/arch/arm/boot/dts/k2l-netcp.dtsi
@@ -282,3 +282,27 @@ netcp: netcp@26000000 {
 		};
 	};
 };
+
+crypto@26080000 {
+	compatible = "ti,keystone-crypto";
+	label = "crypto-gbe";
+	reg = <0x26080000 0x40000>;
+	clocks = <&clksa>;
+
+	ti,navigator-dmas = <&dma_gbe 26>, <&dma_gbe 18>;
+	ti,navigator-dma-names = "crypto-rx", "crypto-tx";
+
+	rx-channel = "crypto-rx";
+	rx-queue-depth = <256 64 0 0>;
+	rx-buffer-size = <1500 4096 0 0>;
+	rx-compl-queue = <536>;
+	rx-pool	= <512 12>;
+
+	tx-channel = "crypto-tx";
+	tx-queue-depth = <256>;
+	tx-compl-queue = <537>;
+	tx-pool = <512 12>;
+	tx-submit-queue = <914>;
+
+	sc-id = <0x7000 0x71ff>;
+};
-- 
1.9.1

